src/components/ClipboardButton/ClipboardButton.tsx

```diff
@@ -1,51 +1,85 @@
 import React, {useEffect, useRef} from 'react';
-import {Button} from '../Button';
+import {Button, ButtonProps} from '../Button';
 import {ClipboardIcon} from '../ClipboardIcon';
 import {CopyToClipboard} from '../CopyToClipboard';
-import {CopyToClipboardBaseProps} from '../CopyToClipboard/types';
+import {CopyToClipboardBaseProps, CopyToClipboardStatus} from '../CopyToClipboard/types';
 import {QAProps} from '../types';
 import {block} from '../utils/cn';
+import {Tooltip, TooltipProps} from '../Tooltip';
 
 import './ClipboardButton.scss';
 
-export interface ClipboardButtonProps extends CopyToClipboardBaseProps, QAProps {
+export interface ClipboardButtonProps
+    extends CopyToClipboardBaseProps,
+        Omit<ClipboardButtonComponentProps, 'status' | 'onClick'>,
+        QAProps {
+    /** Time to restore initial state, ms */
+    timeout?: number;
+}
+
+interface ClipboardButtonComponentProps extends QAProps {
     /** Icon size in pixels */
     size?: number;
     /** Element CSS class */
     className?: string;
-    /** Time to restore initial state, ms */
-    timeout?: number;
+    status: CopyToClipboardStatus;
+    onClick?: ButtonProps['onClick'];
+    /** tooltip component pops */
+    tooltipProps?: ClipboardButtonTooltipProps;
+}
+
+interface ClipboardButtonTooltipProps extends Omit<TooltipProps, 'content' | 'children'> {
+    /** Text shown before copy */
+    startCopyTip?: string;
+    /** Text shown after copy */
+    endCopyTip?: string;
```

Comment:
Do we actually need customization here? I mean all `tooltipProps`. We can add some in the future if it would be requested.

Reply:
It will save our time in future. Almost all future changes in Tooltip will be automatically available here. 

Why is it better to restrict users of using extra props? Let's give them more opportunity to customise  tooltip.